commands/list_keypairs.go

```diff
@@ -23,84 +27,208 @@ var (
+		PreRun: listKeypairsValidationOutput,
+		Run:    listKeypairsOutput,
 	}
 )
 
+// listKeypairsArguments are the actual arguments used to call the
+// listKeypairs() function.
+type listKeypairsArguments struct {
+	apiEndpoint string
+	clusterID   string
+	token       string
+}
+
+// defaultListKeypairsArguments returns a new listKeypairsArguments struct
+// based on global variables (= command line options from cobra).
+func defaultListKeypairsArguments() listKeypairsArguments {
+	return listKeypairsArguments{
+		apiEndpoint: cmdAPIEndpoint,
+		clusterID:   cmdClusterID,
+		token:       cmdToken,
+	}
+}
+
+// listKeypairsResult is the data structure returned by the listKeypairs() function.
+type listKeypairsResult struct {
+	keypairs []gsclientgen.KeyPairModel
+}
+
 func init() {
 	ListKeypairsCommand.Flags().StringVarP(&cmdClusterID, "cluster", "c", "", "ID of the cluster to list key pairs for")
 	ListCommand.AddCommand(ListKeypairsCommand)
 }
 
-func checkListKeypairs(cmd *cobra.Command, args []string) error {
-	if config.Config.Token == "" {
-		return errors.New("You are not logged in. Use '" + config.ProgramName + " login' to log in.")
+// listKeypairsValidationOutput does our pre-checks and shows errors, in case
+// something is missing.
+func listKeypairsValidationOutput(cmd *cobra.Command, extraArgs []string) {
+	args := defaultListKeypairsArguments()
+	err := listKeypairsValidate(&args)
+	if err != nil {
+		var headline string
+		var subtext string
+
+		switch err.Error() {
+		case errNotLoggedIn:
+			headline = "You are not logged in."
+			subtext = "Please log in using 'gsctl login <email>' or set an auth token as a command line argument."
+			subtext += " See `gsctl list keypairs --help` for details."
+		case errClusterIDNotSpecified:
+			headline = "No cluster ID specified."
+			subtext = "Please specify which cluster to list key pairs for, by using the '-c' or '--cluster' argument."
+		default:
+			headline = err.Error()
+		}
+
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
 	}
-	if cmdClusterID == "" {
+}
+
+// listKeypairsValidate validates our pre-conditions and returns an error in
+// case something is missing.
+// If no clusterID argument is given, and a default cluster can be determined,
+// the listKeypairsArguments given as argument will be modified to contain
+// the clusterID field.
+func listKeypairsValidate(args *listKeypairsArguments) error {
+	if config.Config.Token == "" && args.token == "" {
+		return errors.New(errNotLoggedIn)
+	}
+	if args.clusterID == "" {
 		// use default cluster if possible
 		clusterID, _ := config.GetDefaultCluster(requestIDHeader, listKeypairsActivityName, cmdLine, cmdAPIEndpoint)
 		if clusterID != "" {
 			cmdClusterID = clusterID
 		} else {
-			return errors.New("No cluster given. Please use the -c/--cluster flag to set a cluster ID.")
+			return errors.New(errClusterIDNotSpecified)
```

Comment:
We should always mask errors to be able to comprehend their stack in case shit hits the fan.